### PR TITLE
Fix resolution of BAMM 1.0.0 and 2.0.0 models

### LINKED ISSUE #410: [BUG] SAMM-CLI v2.3.1 (also v2.2.2, v2.2.2) BAMM 2.0.0 models referencing other BAMM 2.0.0 models are not properly resolved
**Describe the bug**
Exception thrown for
```
java -jar ~/Downloads/samm-cli-2.3.1.jar  aspect io.catenax.refurbishing_certificate/1.0.0/RefurbishingCertificate.ttl validate
```

when a BAMM 2.0.0 model references another BAMM 2.0.0 model.

stack trace
```
org.eclipse.esmf.aspectmodel.resolver.ModelResolutionException: Could not resolve [urn:bamm:io.openmanufacturing:meta-model:2.0.0#Aspect, urn:bamm:io.openmanufacturing:meta-model:2.0.0#properties, urn:bamm:io.openmanufacturing:meta-model:2.0.0#events, urn:bamm:io.openmanufacturing:meta-model:2.0.0#preferredName, urn:bamm:io.openmanufacturing:meta-model:2.0.0#description, urn:bamm:io.openmanufacturing:meta-model:2.0.0#operations, urn:bamm:io.openmanufacturing:meta-model:2.0.0#characteristic, urn:bamm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic, urn:bamm:io.openmanufacturing:meta-model:2.0.0#Property]
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:176)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:146)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$resolveAspectModel$1(AspectModelResolver.java:126)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at io.vavr.control.Try.flatMap(Try.java:472)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:126)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModel$8(AspectModelResolver.java:363)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at io.vavr.control.Try.flatMap(Try.java:472)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.loadAndResolveModel(AspectModelResolver.java:361)
	at org.eclipse.esmf.AbstractCommand.loadAndResolveModel(AbstractCommand.java:53)
	at org.eclipse.esmf.aspect.AspectValidateCommand.run(AspectValidateCommand.java:60)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at org.eclipse.esmf.LoggingMixin.executionStrategy(LoggingMixin.java:41)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.eclipse.esmf.SammCli.main(SammCli.java:106)
Caused by: java.io.FileNotFoundException: No model file containing urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.shared.recycling_strategy_certificate/1.0.0
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy.apply(FileSystemStrategy.java:93)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy.apply(FileSystemStrategy.java:33)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.getModelForUrn(AspectModelResolver.java:311)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolve(AspectModelResolver.java:243)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:171)
	... 21 more
```

**Where**
Model resolution in SAMM SDK (as error indicates)
it tries to find a urn:samm but should be urn:bam
```
urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.shared.recycling_strategy_certificate/1.0.0

```

Can be reproduced with model
https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.refurbishing_certificate/1.0.0/RefurbishingCertificate.ttl

(repo https://github.com/eclipse-tractusx/sldt-semantic-models.git) needs to be checked out to allow to find the referenced model.

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by atextor**
Fixes #410.

### CODE REVIEW

**Comment by atextor** [comment]
Note: Test failures are due to broken assumptions in diagram generation; will be fixed by #407.

**Review by michelu89 — APPROVED** [review]

**Review by ysrbo — APPROVED** [review]

**Review by Yauhenikapl — APPROVED** [review]

**Review by Yauhenikapl — APPROVED** [review]

**Review by Yauhenikapl — APPROVED** [review]

File: core/esmf-aspect-model-validator/src/test/java/org/eclipse/esmf/aspectmodel/validation/services/AspectModelValidatorTest.java [hunk 1]
```diff
@@ -112,78 +111,40 @@ public void testGetFixForInvalidTestAspectModel( final KnownVersion metaModelVer
...
-                     && (testModel == InvalidTestAspect.ASPECT_MISSING_NAME_AND_PROPERTIES
-                     || testModel == InvalidTestAspect.ASPECT_MISSING_PROPERTIES) ) {
-                  return Stream.of();
-               }
-               return Stream.of( Arguments.of( testModel, metaModelVersion ) );
-            } ) );
+            .filter( invalidTestAspect ->
+                  (!invalidTestAspect.equals( InvalidTestAspect.ASPECT_MISSING_NAME_AND_PROPERTIES )
+                        && !invalidTestAspect.equals( InvalidTestAspect.ASPECT_MISSING_PROPERTIES )) )
+            .flatMap( invalidTestAspect -> Stream.of( Arguments.of( invalidTestAspect )) );
```
Comment (RaMisess) [inline core/esmf-aspect-model-validator/src/test/java/org/eclipse/esmf/aspectmodel/validation/services/AspectModelValidatorTest.java:117]: map( Arguments::of )

Comment (Yauhenikapl) [reply core/esmf-aspect-model-validator/src/test/java/org/eclipse/esmf/aspectmodel/validation/services/AspectModelValidatorTest.java]: It was updated, 

Thank you!

**Comment by Yauhenikapl** [comment]
Hi everyone,

This PR has to cancel. These changes were merged it this PR [#420 ](https://github.com/eclipse-esmf/esmf-sdk/pull/420).

Thanks,
Best Regards!

**Pull request closed without merging**